pkg/grpc/requestinfo/requestinfo.go

```diff
@@ -335,12 +348,37 @@ func sourceAddr(ctx context.Context) net.Addr {
 	return p.Addr
 }
 
-func metadataFromHeader(header http.Header) metadata.MD {
-	md := make(metadata.MD)
-	for key, vals := range header {
-		md.Append(key, vals...)
+// WithGet adds the HeaderGetter implementation to http.Header.
+type WithGet http.Header
```

Comment:
```suggestion
type AsHeaderGetter http.Header
```
or `AsHeadersMultiMap` if you go with the other suggestion. Because `http.Header` has `Get()` already.